pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -212,19 +221,20 @@ func (gb *GraphBuilder) syncMonitors(resources map[schema.GroupVersionResource]s
 			errs = append(errs, fmt.Errorf("couldn't look up resource %q: %v", resource, err))
 			continue
 		}
-		c, s, err := gb.controllerFor(resource, kind)
+		c, s, d, err := gb.controllerFor(resource, kind)
 		if err != nil {
 			errs = append(errs, fmt.Errorf("couldn't start monitor for resource %q: %v", resource, err))
 			continue
 		}
-		current[resource] = &monitor{store: s, controller: c}
+		current[resource] = &monitor{store: s, controller: c, destroy: d}
 		added++
 	}
 	gb.monitors = current
 
 	for _, monitor := range toRemove {
 		if monitor.stopCh != nil {
 			close(monitor.stopCh)
+			monitor.destroy()
```

Comment:
don't we want to call this even if stopCh is nil? we already registered the handler, even if we didn't call startMonitors() and set the stopCh

Reply:
yes, we need to remove whether monitor start or not. fixed